.github/workflows/integration_tests.yml

```diff
@@ -95,14 +94,26 @@ jobs:
           key: ${{ runner.os }}-multi-buildx-${{ github.sha }}
           restore-keys: |
             ${{ runner.os }}-multi-buildx
+      - name: container-login
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: labels
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.IMAGE_NAME }}
       - name: Make upload dir
         run: mkdir -p ${{ env.UPLOAD_DIR }}
       - name: build-and-push
         uses: docker/build-push-action@v2
         with:
           file: Dockerfile
-          push: false
-          tags: steward:prebuilt
+          push: ${{ github.event_name != 'pull_request' }}
```

Comment:
I suppose my main question here is: do we need to push to Docker at all for steward and hardhat images in our CI workflows? Couldn't we just build the steward and hardhat docker images as artifacts and remove the push aspect entirely?

Reply:
my thinking was we wanted the ‘steward:main‘ tag to get pushed when merging to main?

Reply:
We don't really need to publish any docker images unless something else needs to pull them. Steward as a repo is effectively a leaf node in the dependency tree.